pkg/client/typed_client.go

```diff
@@ -156,6 +156,32 @@ func (c *typedClient) UpdateStatus(ctx context.Context, obj runtime.Object) erro
 		Name(o.GetName()).
 		SubResource("status").
 		Body(obj).
+		VersionedParams((&UpdateOptions{}).ApplyOptions(opts).AsUpdateOptions(), c.paramCodec).
+		Context(ctx).
+		Do().
+		Into(obj)
+}
+
+// PatchStatus used by StatusWriter to write status.
+func (c *typedClient) PatchStatus(ctx context.Context, obj runtime.Object, patch Patch, opts ...PatchOptionFunc) error {
+	o, err := c.cache.getObjMeta(obj)
+	if err != nil {
+		return err
+	}
+
+	data, err := patch.Data(obj)
+	if err != nil {
+		return err
+	}
+
+	patchOpts := &PatchOptions{}
+	return o.Patch(patch.Type()).
+		NamespaceIfScoped(o.GetNamespace(), o.isNamespaced()).
+		Resource(o.resource()).
+		Name(o.GetName()).
+		SubResource("status").
```

Comment:
ok (I get it) server-side apply is implemented using patch on status sub-resource, little intuitive but I am sure they must have some reasons for it. 
So that means, CRD needs to have status sub-resource enabled (I wonder if we should generate marker for status subresource by default or we already do ?)

Reply:
> ok (I get it) server-side apply is implemented using patch on status sub-resource, little intuitive [clipped]

Not just status, all resources/sub-resources.